### ISSUE TITLE
Translations for Config Ui X

### ISSUE THREAD

**Issue by razer4908** [oznu/homebridge-config-ui-x#102]
Maybe its possible to select an Language file to Config Ui. Would be very nice

**Comment by oznu** [comment]
I've been thinking about this for a while, this library looks like it will make it easier:

https://github.com/ngx-translate/core

I'll create the english language files, but I will then need help from others to create translations.

**Comment by razer4908** [comment]
I can translate it to German if you like

**Comment by palasinio** [comment]
I can help with German, too...

**Comment by oznu** [comment]
Working branch:

https://github.com/oznu/homebridge-config-ui-x/tree/enable-i18n

**Comment by oznu** [comment]
Thanks @razer4908 for your help so far fixing up Google Translate's attempt at english to german. A pre-release can installed using this command (don't forget to restart homebridge after you do this):

```
sudo npm install -g --unsafe-perm homebridge-config-ui-x@next
```

There is nothing to configure, the app will figure out what language to serve based on your browser settings. When this gets made live you can upgrade in-app as usual.

German translation file: https://github.com/oznu/homebridge-config-ui-x/blob/enable-i18n/ui/src/i18n/de.json

**Comment by the0neyouseek** [comment]
Hi @oznu,

First, let me thank you for your work. 
I've took the liberty to make a pr (#105) to add French translation files.

Tell me if it's good or need re-work.
Have a nice day :smile:

**Comment by oznu** [comment]
Landed in 3.8.0.

**Comment by YellowNest** [comment]
I would like to bump this thread because I have a suggestion on how to work more easily with the translation of the homebridge ui. I'm involved in some translation jobs for various projects out there and many projects use POEditor - a really awesome tool that makes translation jobs a lot easier! Submit this suggestion to the creator!
